CONTRIBUTING.md

```diff
@@ -0,0 +1,192 @@
+
+## Table of Contents
+
+- [Contributor Onboarding](#contributor-onboarding)
+  - [Table of Contents](#table-of-contents)
+  - [What should I know before I get started](#what-should-i-know-before-i-get-started)
+  - [Pre-requisites](#pre-requisites)
+  - [Pull Request Change flow](#pull-request-change-flow)
+  - [Development Setup](#development-setup)
+    - [Visual Studio Code Extensions](#visual-studio-code-extensions)
+    - [Setting up the  end-to-end debugging environment](#setting-up-the-end-to-end-debugging-environment)
+  - [Pre Commit Tasks](#pre-commit-tasks)
+    - [Running unit tests](#running-unit-tests)
+  - [Continuous Integration Guidelines & Conventions](#continuous-integration-guidelines--conventions)
+  - [Getting help](#getting-help)
+    - [Requesting a release](#requesting-a-release)
+
+## What should I know before I get started
+
+- [Azure Functions Java Quickstart](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function-azure-cli?tabs=bash%2Cbrowser&pivots=programming-language-java)
+- [Azure Function Java developer guide](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-java?tabs=consumption)
+
+## Pre-requisites
+
+- OS
+  - MacOS, Ubuntu (or) Windows10
+- IDE
+  - [IntelliJ](https://www.jetbrains.com/idea/download)
+  - [Java extension in VS Code](https://code.visualstudio.com/docs/languages/java)
+- Java Tools
+  - [Maven](https://maven.apache.org/install.html)
+  - [JDK (Azul Zulu for Azure) 8](https://www.azul.com/downloads/azure-only/zulu/?version=java-8-lts&architecture=x86-64-bit&package=jdk) - this also installs Java 1.8 JRE
+- Azure Tools
+  - [Azure Storage Emulator](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator) (or) [Create a storage account in Azure](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-portal)
+  - [Azure Functions Core Tools](https://github.com/Azure/azure-functions-core-tools) v2.7.x and above.
+  - [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/)
+  
+## Pull Request Change flow
+
+The general flow for making a change to the library is:
+
+1. 🍴 Fork the [worker](https://github.com/helayoty/azure-functions-java-worker) repo (add the fork via `git remote add me <clone url here>`
+2. 🌳 Create a branch for your change (generally branch from dev) (`git checkout -b my-change`)
+3. 🛠 Make your change
+4. ✔️ Test your change
+5. ⬆️ Push your changes to your fork (`git push me my-change`)
+6. 💌 Open a PR to the dev branch
+7. 📢 Address feedback and make sure tests pass (yes even if it's an "unrelated" test failure)
```

Comment:
For the current CI, The forked repo users need to ask us to run the CI pipeline. Then I or Ahmed will run the pipeline.